addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaBridgeHandler.java

```diff
@@ -128,49 +147,59 @@ private void setupRefreshTimer(int initialWaitTime) {
         if (refreshTimer != null) {
             refreshTimer.cancel(false);
         }
-        refreshTimer = scheduler.scheduleWithFixedDelay(new Runnable() {
-            @Override
-            public void run() {
-                updateAllZoneInformation();
-            }
-        }, initialWaitTime, refrehInterval, TimeUnit.SECONDS);
+        refreshTimer = scheduler.scheduleWithFixedDelay(() -> updateAllZoneInformation(), initialWaitTime,
+                refrehInterval, TimeUnit.SECONDS);
     }
 
     /**
-     * Periodically and initially called. Updates
+     * Periodically and initially called. This must run in another thread, because
+     * all update calls are blocking.
      */
     void updateAllZoneInformation() {
+        logger.trace("updateAllZoneInformation");
         try {
-            SystemControl basicDeviceInformation = new SystemControl();
-            basicDeviceInformation.fetchDeviceInformation(xml, state);
-            basicDeviceInformation.fetchPowerInformation(xml, state);
-
-            updateProperty(YamahaReceiverBindingConstants.PROPERTY_VERSION, state.version);
-            updateProperty(YamahaReceiverBindingConstants.PROPERTY_ASSIGNED_NAME, state.name);
-
+            DeviceInformation deviceInformation = ProtocolFactory.DeviceInformation(connection, deviceInformationState);
+            deviceInformation.update();
+            zoneDiscoveryService.publishZones(deviceInformationState, thing.getUID());
+
+            SystemControl systemControl = ProtocolFactory.SystemControl(connection, this);
+            systemControlState.power = true;
+            systemControl.update();
+
+            updateProperty(YamahaReceiverBindingConstants.PROPERTY_VERSION, deviceInformationState.version);
+            updateProperty(YamahaReceiverBindingConstants.PROPERTY_ASSIGNED_NAME, deviceInformationState.name);
+
+            updateStatus(ThingStatus.ONLINE);
+
+            Bridge bridge = (Bridge) thing;
+            List<Thing> things = bridge.getThings();
+            for (Thing thing : things) {
```

Comment:
C++ habit. In C++ `bridge.getThings()` would be called in every iteration under some circumstances and if it is a heavy operation (like computing the list every time), it is a better option to cache the variable. In Java, an Iterator would be used internally I guess, so there is no performance impact.

@zarusz Can you change it to `for (Thing thing : ((Bridge) thing).getThings()) {`?